postgres-appliance/build_scripts/base.sh

```diff
@@ -129,22 +129,41 @@ for version in $DEB_PG_SUPPORTED_VERSIONS; do
 
     # Install 3rd party stuff
 
-    # use subshell to avoid having to cd back (SC2103)
-    (
-        cd timescaledb
-        for v in $TIMESCALEDB; do
-            git checkout "$v"
-            sed -i "s/VERSION 3.11/VERSION 3.10/" CMakeLists.txt
-            if BUILD_FORCE_REMOVE=true ./bootstrap -DREGRESS_CHECKS=OFF -DWARNINGS_AS_ERRORS=OFF \
-                    -DTAP_CHECKS=OFF -DPG_CONFIG="/usr/lib/postgresql/$version/bin/pg_config" \
-                    -DAPACHE_ONLY="$TIMESCALEDB_APACHE_ONLY" -DSEND_TELEMETRY_DEFAULT=NO; then
-                make -C build install
-                strip /usr/lib/postgresql/"$version"/lib/timescaledb*.so
-            fi
-            git reset --hard
-            git clean -f -d
-        done
-    )
+    # Used to avoid installing toolkit if timescaledb wasn't installed for the pgversion.
+    __timescaleDbInstalled="false"
+
+    pushd .
+    cd timescaledb
+    for v in $TIMESCALEDB; do
+        git checkout "$v"
+        sed -i "s/VERSION 3.11/VERSION 3.10/" CMakeLists.txt
+        if BUILD_FORCE_REMOVE=true ./bootstrap -DREGRESS_CHECKS=OFF -DWARNINGS_AS_ERRORS=OFF \
+                -DTAP_CHECKS=OFF -DPG_CONFIG="/usr/lib/postgresql/$version/bin/pg_config" \
+                -DAPACHE_ONLY="$TIMESCALEDB_APACHE_ONLY" -DSEND_TELEMETRY_DEFAULT=NO; then
+            make -C build install
+            strip /usr/lib/postgresql/"$version"/lib/timescaledb*.so
+            __timescaleDbInstalled="true"
+        fi
+        git reset --hard
+        git clean -f -d
+    done
+    popd
+
+    if [ "${TIMESCALEDB_APACHE_ONLY}" != "true" ] && [ "${TIMESCALEDB_TOOLKIT}" = "true" ] && [ "${__timescaleDbInstalled}" = "true" ]; then
+        __versionCodename=$(sed </etc/os-release -ne 's/^VERSION_CODENAME=//p')
+        echo "deb [signed-by=/usr/share/keyrings/timescale_E7391C94080429FF.gpg] https://packagecloud.io/timescale/timescaledb/ubuntu/ ${__versionCodename} main" | tee /etc/apt/sources.list.d/timescaledb.list
+        curl -L https://packagecloud.io/timescale/timescaledb/gpgkey | apt-key --keyring /usr/share/keyrings/timescale_E7391C94080429FF.gpg add -
+
+        apt-get update
+        if [ $(apt-cache search --names-only "^timescaledb-toolkit-postgresql-${version}$" | wc -l) -eq 1 ]; then
+            apt-get install "timescaledb-toolkit-postgresql-$version"
```

Comment:
Using `apt-cache` to check for the exact package name as I didn't want to swallow errors from `apt-get install`, by assuming errors from it means package not found.

I'm making the assumption here that if `apt-cache` outputs a single line that means the package is found.
I think this is okay, but it is a little nasty and if you prefer I can change from `wc -l` to instead grep for the package name in the output.

Reply:
I think it is fine for now